src/iarc7_motion/iarc_tasks/hit_roomba_task.py

```diff
@@ -0,0 +1,219 @@
+                if not self._check_max_roomba_range():
+                    return (TaskAborted(msg='The provided roomba is not close enough to the quad'),)
+                
+                roomba_x_velocity = self._roomba_odometry.twist.twist.linear.x
+                roomba_y_velocity = self._roomba_odometry.twist.twist.linear.y
+
+                # p-controller
+                x_vel_target = (self._roomba_point.point.x * self._K_X + roomba_x_velocity)
+                y_vel_target = (self._roomba_point.point.y * self._K_Y + roomba_y_velocity)
+                
+                z_vel_target = self._descent_velocity
+
+                #caps velocity
+                vel_target = math.sqrt(x_vel_target**2 + y_vel_target**2)
+
+                if vel_target > self._MAX_HORIZ_SPEED:
+                    x_vel_target = x_vel_target * (self._MAX_HORIZ_SPEED/vel_target)
+                    y_vel_target = y_vel_target * (self._MAX_HORIZ_SPEED/vel_target)
+                
+                if (abs(z_vel_target) > self._MAX_Z_VELOCITY):
+                    z_vel_target = z_vel_target/abs(z_vel_target) * self._MAX_Z_VELOCITY
+
+                desired_vel = [x_vel_target, y_vel_target, z_vel_target]
+
+                drone_vel_x = self._drone_odometry.twist.twist.linear.x
+                drone_vel_y = self._drone_odometry.twist.twist.linear.y
+                drone_vel_z = self._drone_odometry.twist.twist.linear.z
+
+                curr_vel = [drone_vel_x, drone_vel_y, drone_vel_z]
+
+                desired_vel = self._limiter.limit_acceleration(curr_vel, desired_vel)
+
+                velocity = TwistStamped()
+                velocity.header.frame_id = 'level_quad'
+                velocity.header.stamp = rospy.Time.now()
+                velocity.twist.linear.x = desired_vel[0]
+                velocity.twist.linear.y = desired_vel[1]
+                velocity.twist.linear.z = desired_vel[2]
+                
+                return (TaskRunning(), VelocityCommand(velocity))
+
+            else:
+                return (TaskAborted(msg='Illegal state reached in Hit Roomba Task' ),)
+
+    # checks to see if passed in roomba id is in sight of quad
+    def _check_roomba_in_sight(self):
+        for odometry in self._roomba_array.data:
+            if odometry.child_frame_id == self._roomba_id:
+                self._roomba_odometry = odometry
+                return True
+        return False
+
+    # that the drone and roomba are both within a specified distance
+    # in order to start/continue the task
+    def _check_max_roomba_range(self):
+        _distance_to_roomba = math.sqrt(self._roomba_point.point.x**2 + 
+                            self._roomba_point.point.y**2)
+        
+        return (_distance_to_roomba <= self._MAX_START_TASK_DIST)
+
+    def cancel(self):
+        rospy.loginfo('HitRoomba Task canceled')
+        self._canceled = True
+
+    def _on_ground(self):
```

Comment:
This won't necessarily be true if we land catywonkus on the roomba, but this can easily be fixed after testing.